### ISSUE TITLE
0.7.0: build fails

### ISSUE THREAD

**Issue by kloczek** [Unrud/video-downloader#78]
```
+ /usr/bin/meson compile -C noarch-redhat-linux-gnu -j 48 --verbose
Found runner: ['/usr/bin/ninja']
ninja: Entering directory `noarch-redhat-linux-gnu'
[1/10] /usr/bin/glib-compile-resources ../src/video_downloader.gresource.xml --sourcedir ../src --internal --generate --target src/video-downloader.gresource --dependency-file src/video-downloader.gresource.d
[2/10] /usr/bin/rsvg-convert --height 16 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_16.png ../data/com.github.unrud.VideoDownloader.svg
FAILED: data/com.github.unrud.VideoDownloader_16.png
/usr/bin/rsvg-convert --height 16 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_16.png ../data/com.github.unrud.VideoDownloader.svg
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[3/10] /usr/bin/rsvg-convert --height 32 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_32.png ../data/com.github.unrud.VideoDownloader.svg
FAILED: data/com.github.unrud.VideoDownloader_32.png
/usr/bin/rsvg-convert --height 32 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_32.png ../data/com.github.unrud.VideoDownloader.svg
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[4/10] /usr/bin/rsvg-convert --height 48 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_48.png ../data/com.github.unrud.VideoDownloader.svg
FAILED: data/com.github.unrud.VideoDownloader_48.png
/usr/bin/rsvg-convert --height 48 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_48.png ../data/com.github.unrud.VideoDownloader.svg
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[5/10] /usr/bin/rsvg-convert --height 64 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_64.png ../data/com.github.unrud.VideoDownloader.svg
FAILED: data/com.github.unrud.VideoDownloader_64.png
/usr/bin/rsvg-convert --height 64 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_64.png ../data/com.github.unrud.VideoDownloader.svg
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[6/10] /usr/bin/rsvg-convert --height 128 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_128.png ../data/com.github.unrud.VideoDownloader.svg
FAILED: data/com.github.unrud.VideoDownloader_128.png
/usr/bin/rsvg-convert --height 128 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_128.png ../data/com.github.unrud.VideoDownloader.svg
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[7/10] /usr/bin/rsvg-convert --height 256 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_256.png ../data/com.github.unrud.VideoDownloader.svg
FAILED: data/com.github.unrud.VideoDownloader_256.png
/usr/bin/rsvg-convert --height 256 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_256.png ../data/com.github.unrud.VideoDownloader.svg
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[8/10] /usr/bin/rsvg-convert --height 512 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_512.png ../data/com.github.unrud.VideoDownloader.svg
FAILED: data/com.github.unrud.VideoDownloader_512.png
/usr/bin/rsvg-convert --height 512 --keep-aspect-ratio --format png --output data/com.github.unrud.VideoDownloader_512.png ../data/com.github.unrud.VideoDownloader.svg
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[9/10] /usr/bin/meson --internal msgfmthelper ../data/com.github.unrud.VideoDownloader.desktop.in data/com.github.unrud.VideoDownloader.desktop desktop ../data/../po
[10/10] /usr/bin/meson --internal msgfmthelper ../data/com.github.unrud.VideoDownloader.appdata.xml.in data/com.github.unrud.VideoDownloader.appdata.xml xml ../data/../po
ninja: build stopped: subcommand failed.
```

**Comment by Unrud** [comment]
**rsvg-convert** fails when converting *data/com.github.unrud.VideoDownloader.svg* to PNG. The SVG image wasn't changed in version 0.7.0.

**Comment by kloczek** [comment]
And should be updated?
0.6.8 in the same build env was OK.

**Comment by Unrud** [comment]
> And should be updated?

I think the problem was introduced by an update of **rsvg-convert**.
What version are you using?

**Comment by kloczek** [comment]
latest .. 2.51.0

**Comment by Unrud** [comment]
> Latest one .. 2.51.0

I've compiled this version from source and it works without errors.

```
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/tkloczko/rpmbuild/BUILD/librsvg-2.51.0/vendor/clap/src/app/parser.rs:1454:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The error message indicates that the crash appears in https://github.com/clap-rs/clap, which is the external package for parsing command line arguments used by **rsvg-convert**.

Maybe your version of **rsvg-convert** was compiled with a buggy or incompatible version.

**Comment by kloczek** [comment]
0.7.1 seems still has the same issue with test suite ..

I'm compiling al my stuff amd I don't see any clap build dependencies in librsv2 :/
I dont see anything like clap in cargo or rust packages
```
# rpm -ql cargo rust | grep clap
#
```
In other words I'm not sure about what you are talking .. (may I ask fo some explanation?)

**Comment by GustavoPeredo** [comment]
No problem here either... @kloczek have you tried reinstalling rsvg-compile from your distro repositories?

**Comment by kloczek** [comment]
```
[root@barrel Packages]# rpm -qf /usr/bin/rsvg-convert
librsvg2-tools-2.51.0-2.fc34.x86_64
[root@barrel Packages]# rpm -V librsvg2-tools-2.51.0-2.fc34.x86_64
[root@barrel Packages]# dnf reinstall librsvg2-tools
Last metadata expiration check: 1:07:08 ago on Tue 09 Feb 2021 03:41:59 GMT.
Dependencies resolved.
============================================================================================================================================================================
 Package                                       Architecture                          Version                                       Repository                          Size
============================================================================================================================================================================
Reinstalling:
 librsvg2-tools                                x86_64                                2.51.0-2.fc34                                 g2v                                1.9 M

Transaction Summary
============================================================================================================================================================================

Total download size: 1.9 M
Installed size: 5.9 M
Is this ok [y/N]: y
Downloading Packages:
librsvg2-tools-2.51.0-2.fc34.x86_64.rpm                                                                                                      43 MB/s | 1.9 MB     00:00
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                        41 MB/s | 1.9 MB     00:00
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                    1/1
  Reinstalling     : librsvg2-tools-2.51.0-2.fc34.x86_64                                                                                                                1/2
  Cleanup          : librsvg2-tools-2.51.0-2.fc34.x86_64                                                                                                                2/2
  Running scriptlet: librsvg2-tools-2.51.0-2.fc34.x86_64                                                                                                                2/2
  Verifying        : librsvg2-tools-2.51.0-2.fc34.x86_64                                                                                                                1/2
  Verifying        : librsvg2-tools-2.51.0-2.fc34.x86_64                                                                                                                2/2

Reinstalled:
  librsvg2-tools-2.51.0-2.fc34.x86_64

Complete!
```

**Comment by Unrud** [comment]
> In other words I'm not sure about what you are talking .. (may I ask fo some explanation?)

https://gitlab.gnome.org/GNOME/librsvg/-/blob/588dec14c980e2dcd0c00a03aac65abdf5266349/Cargo.toml#L20

**Comment by kloczek** [comment]
With 0.7.2 build seems is OK but test suite is failing
```
+ /usr/bin/meson test -C noarch-redhat-linux-gnu --num-processes 48 --print-errorlogs
ninja: Entering directory `/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu'
ninja: no work to do.
1/4 Validate desktop file          OK              0.03s
2/4 Validate schema file           OK              0.02s
3/4 Lint python files              FAIL            0.51s   exit status 1
>>> MALLOC_PERTURB_=4 /usr/bin/python3 -m flake8 /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/__init__.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/about_dialog.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/authentication_dialog.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/main.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/model.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/playlist_dialog.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/util.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/window.py /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/src/video-downloader
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/authentication_dialog.py:20:34: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/main.py:21:30: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/main.py:24:63: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/main.py:25:43: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/model.py:24:30: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/model.py:28:48: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/playlist_dialog.py:20:34: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/util.py:21:31: H301: one import per line
/home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/../src/window.py:23:36: H301: one import per line
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

4/4 Validate appstream file        OK              0.79s

Summary of Failures:

3/4 Lint python files       FAIL            0.51s   exit status 1


Ok:                 3
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0

Full log written to /home/tkloczko/rpmbuild/BUILD/video-downloader-0.7.2/noarch-redhat-linux-gnu/meson-logs/testlog.txt
```

**Comment by Unrud** [comment]
> With 0.7.2 build seems is OK but test suite is failing

Should be fixed in d995d1e4b08da479a031beaa5df476b8fdb83dca
